replication-api-impl/src/test/java/com/connexta/replication/api/impl/worker/WorkerManagerComponentTest.java

```diff
@@ -118,43 +151,139 @@ public void cleanup() throws Exception {
   }
 
   @Test
-  public void testWorkerTakesFromQueue() throws Exception {
+  public void testWorkerHarvestDdfResource() throws Exception {
+    // setup
+    final URI resourceUri = new URI("http://remote.test:1234/testId");
+    final Date modifiedDate = new Date();
+    final int resourceSize = 10;
+    final Metadata metadata = createMetadata(METADATA_ID, resourceUri, modifiedDate, resourceSize);
+    final ResourceInfo resourceInfo = createResourceInfo(metadata);
+    final MetadataInfo metadataInfo = createMetadataInfo(metadata);
     MemorySiteQueue queue = (MemorySiteQueue) broker.getQueue(REMOTE_SITE_ID);
-    TaskInfo taskInfo1 = createTaskInfo("taskId1");
-    TaskInfo taskInfo2 = createTaskInfo("taskId2");
-    TaskInfo taskInfo3 = createTaskInfo("taskId3");
-    queue.put(taskInfo1);
-    queue.put(taskInfo2);
-    queue.put(taskInfo3);
+    TaskInfo taskInfo = createTaskInfo(METADATA_ID, metadataInfo, resourceInfo);
+    Task task = queue.putAndPeek(taskInfo);
 
+    ArgumentCaptor<CreateStorageRequest> createStorageRequestCaptor = ArgumentCaptor
```

Comment:
I can't wait to have our endpoint stubs so we can replace all that mocking with simple checks of what they received.